src/main/java/com/microsoft/azure/spark/tools/http/HttpObservable.java

```diff
@@ -335,20 +338,20 @@ public HttpObservable setTrustStrategy(final TrustStrategy externalTrustStrategy
     /**
      * Helper to convert the http response to a specified type.
      *
+     * @param <T> the target type
      * @param resp HTTP response, consumed as String content
      * @param clazz the target type to convert
-     * @param <T> the target type
      * @return the specified type class instance
      */
-    public <T> T convertJsonResponseToObject(final HttpResponse resp, final Class<T> clazz) {
+    public <T> Pair<T, HttpResponse> convertJsonResponseToObject(final HttpResponse resp, final Class<T> clazz) {
```

Comment:
This is a pretty big change, lots of codes will be affected. Could you help to explain why we need this change?